lib/buffer.js

```diff
@@ -97,13 +114,22 @@ function alignPool() {
  **/
 function Buffer(arg, encodingOrOffset, length) {
   // Common case.
+  if (pendingDeprecation && bufferWarn) {
+    // This is a *pending* deprecation warning. It is not emitted by
+    // default unless the --pending-deprecation command-line flag is
+    // used or the NODE_PENDING_DEPRECATION=1 envvar is set.
+    process.emitWarning(bufferWarning, 'DeprecationWarning', 'DEP0005');
+    bufferWarn = false;
+  }
+
   if (typeof arg === 'number') {
     if (typeof encodingOrOffset === 'string') {
       throw new Error(
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    assertSize(arg);
+    return new FastBuffer(createRandomFillBuffer(arg));
```

Comment:
Yeah. I'm leaning in that direction to be honest (specifically, using `Buffer.alloc(size, fill)`). It would be much less disruptive and wouldn't require the underlying c/c++ change. I did it this way first to see what the perf would be like and it is way too much of a hit